packages/picasso.js/src/core/chart-components/ref-line/lines-and-labels.js

```diff
@@ -61,46 +150,55 @@ export function alignmentToNumber(align) {
  * @param {object[]} items - Array of all items (for collision detection)
  * @ignore
  */
-export function createLineWithLabel({ chart, blueprint, renderer, p, settings, items }) {
+export function createLineWithLabel({ chart, blueprint, renderer, p, settings, items, slopeLine }) {
   let doesNotCollide = true;
   let line = false;
+  let slope = false;
   let rect = false;
   let label = false;
   let value = false;
   let style = extend(true, {}, settings.style.line, p.line || {});
+  let slopeStyle = extend(true, refLabelDefaultSettings(), settings.style.label || {}, { fill: style.stroke });
 
   // Use the transposer to handle actual positioning
-  line = blueprint.processItem({
-    type: 'line',
-    x1: p.position,
-    y1: 0,
-    x2: p.position,
-    y2: 1,
-    stroke: style.stroke || 'black',
-    strokeWidth: style.strokeWidth || 1,
-    strokeDasharray: style.strokeDasharray,
-    flipXY: p.flipXY || false, // This flips individual points (Y-lines)
-    value: p.valueInfo ? p.valueInfo.id : p.value,
-  });
+  if (slopeLine) {
+    slope = blueprint.processItem({
+      type: 'line',
+      x1: slopeLine.x1,
+      y1: slopeLine.y1,
+      x2: slopeLine.x2,
+      y2: slopeLine.y2,
+      stroke: style.stroke || 'black',
+      strokeWidth: style.strokeWidth || 1,
+      strokeDasharray: style.strokeDasharray,
+      flipXY: false,
+      value: p.valueInfo ? p.valueInfo.id : p.value,
+    });
+  } else {
+    line = blueprint.processItem({
+      type: 'line',
+      x1: p.position,
+      y1: 0,
+      x2: p.position,
+      y2: 1,
+      stroke: style.stroke || 'black',
+      strokeWidth: style.strokeWidth || 1,
+      strokeDasharray: style.strokeDasharray,
+      flipXY: p.flipXY || false, // This flips individual points (Y-lines)
+      value: p.valueInfo ? p.valueInfo.id : p.value,
+    });
+  }
 
   if (p.label) {
-    const item = extend(true, refLabelDefaultSettings(), settings.style.label || {}, { fill: style.stroke }, p.label);
+    let item = extend(true, refLabelDefaultSettings(), settings.style.label || {}, { fill: style.stroke }, p.label);
```

Comment:
Why change this to a `let` declaration? The variable does not appear to be reassigned